pypinfo/core.py

```diff
@@ -12,13 +12,17 @@
 FROM
   TABLE_DATE_RANGE(
     [the-psf:pypi.downloads],
-    DATE_ADD(CURRENT_TIMESTAMP(), {}, "day"),
-    DATE_ADD(CURRENT_TIMESTAMP(), {}, "day")
+    {},
+    {}
   )
 """
+DATE_ADD = 'DATE_ADD(CURRENT_TIMESTAMP(), {}, "day")'
+START_TIMESTAMP = 'TIMESTAMP("{} 00:00:00")'
+END_TIMESTAMP = 'TIMESTAMP("{} 23:59:59")'
 START_DATE = '-31'
 END_DATE = '-1'
 DEFAULT_LIMIT = '10'
+YYYY_MM_DD = re.compile("^[0-9]{4}-[01][0-9]-[0-3][0-9]$")
```

Comment:
Do you think it would be valuable to further enforce these?

month - `([0][1-9]|[1][0-2])`
year - `([0][1-9]|[12][0-9]|[3][01])`

Reply:
I considered something like that, then thought a basic check is enough to make sure the user is using the right sort of string, eg. not `--start-date "14 May 2018"`.

But BigQuery raises this exception with `--start-date 2018-19-39`:
```
google.api_core.exceptions.BadRequest: 400 Timestamp evaluation: result is NaN or NULL
```

Which isn't very clear. And you know what they say about solving things with regexes -- "now you have two problems" -- so perhaps this would be more robust? https://stackoverflow.com/a/16870699/724176